pkg/network/kubemacpool.go

```diff
@@ -95,3 +109,21 @@ func generateRandomMacPrefix() ([]byte, error) {
 
 	return prefix, nil
 }
+
+func checkRange(startMac, endMac net.HardwareAddr) error {
+	for idx := 0; idx <= 5; idx++ {
+		if startMac[idx] < endMac[idx] {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid range. Range end is lesser than its start. start: %v end: %v", startMac, endMac)
+}
+
+func checkCast(mac net.HardwareAddr) error {
+	castBit := 1 & mac[0]
+	if castBit == 1 {
+		return fmt.Errorf("invalid mac address. Multicast addressing is not supported. Should be unicast addressing. The first octet is %#0X", mac[0])
+	}
+
```

Comment:
Yes but after a discussion we can't find a reason to block the user if he want to allocate a vendor reserve range. we just going to block if the range is a multicast one.

when we generate the range it will always be a local one.

Reply:
Fair enough.

Reply:
We discussed about this at the CNV network sync , and we concluded that we should allow both local and universal .